third_party/move/move-compiler-v2/src/pipeline/ability_processor.rs

```diff
@@ -196,7 +197,9 @@ impl<'a> TransferFunctions for CopyDropAnalysis<'a> {
                 // point, is again used in the argument list. Also, in difference to assign inference, we only need
                 // to copy the argument if its not primitive.
                 for (i, src) in srcs.iter().enumerate() {
-                    if (temp_needs_copy(src) || srcs[i + 1..].contains(src)) && type_needs_copy(src)
+                    if !live_srcs_in_dests.contains(src)
+                        && (temp_needs_copy(src) || srcs[i + 1..].contains(src))
```

Comment:
Please keep this code clean:

1. `temp_needs_copy` is the single definition which determines whether we need to copy or not. Logic should go there.
2. You really should have a a function `livevar.is_temp_used_after(temp, instr)` and use that in the definition of `temp_needs_copy`.
3. `fn is_temp_used_after(&self, instr) { self.after.contains_key(temp) && !instrs.dests().contains(temp) }`

No need to compute the live_srcs_in_dests.

Reply:
done